### ISSUE TITLE
在 SecretFlow 中实现简单场景下两方拆分 ESMM 算法

### ISSUE THREAD

**Issue by Candicepan** [secretflow/secretflow#1006]
> 此 ISSUE 为 [隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP）Phase 4 ](https://studio.secretflow.com/activity/fy3n4tddfe3xteq/detail)任务 ISSUE，欢迎社区开发者参与共建～
> - 认领前，辛苦确认是否完成[报名](https://studio.secretflow.com/activity/fy3n4tddfe3xteq/detail)～
> - 更多任务，可查看 [「OSCP Phase4 Season of Dev」Project](https://github.com/orgs/secretflow/projects/11)
>
> This ISSUE is one of the tasks of the  [SecretFlow Open Source Contribution Plan (referred to as SF OSCP) Phase 4](https://studio.secretflow.com/activity/fy3n4tddfe3xteq/detail). Welcome to join us in building it together!
> - Before claiming a task, please make sure you have [signed up](https://studio.secretflow.com/activity/fy3n4tddfe3xteq/detail).
> - For more tasks, you can check the ["OSCP Phase4 Season of Dev" Project](https://github.com/orgs/secretflow/projects/11).

## 任务介绍
- **任务名称**：在 SecretFlow 中实现复杂场景下两方拆分 ESMM 算法
- **技术方向**：FL/application
- **任务难度**：挑战🌟🌟🌟
- **任务期望时间**：7 周

## 详细要求
- 安全性：尽量降低数据、模型泄露的可能
- 功能性：在 SecretFlow 中实现两方拆分版本的 ESMM 算法，参考论文[Entire Space Multi-Task Model: An Eﬀective Approach for Estimating Post-Click Conversion Rate](https://arxiv.org/abs/1804.07931)。
- 收敛性：跑出实验数据并且证明收敛性
- 特征交叉：双方特征可以各自交叉，特征全局交叉是挑战性要求
- 目标拆分：ctr、cvr目标各位于一方
- 代码规范：Python 代码需要使用 black+isort 进行格式化（流水线包含代码规范检查卡点）
- 框架后端：实现 PyTorch 版本。若在完成 PyTorch 版本实现的基础上同步进行 TensorFlow 版本实现将额外累积一颗🌟
- 提交说明：关联该 issue 并提交代码至 https://github.com/secretflow/secretflow/tree/main/secretflow_fl/ml/nn/applications

## 能力要求
- 了解基本 git 操作
- 熟悉 深度学习、联邦学习
- 对 SecretFlow 有一定了解

## 操作说明
- 操作指引：https://www.secretflow.org.cn/docs/secretflow/latest/zh-Hans/tutorial/sl_rec_deepfm
- 范例：https://github.com/secretflow/secretflow/blob/main/secretflow_fl/ml/nn/applications/sl_bst_torch.py

**Comment by zhaocaibei123** [comment]
> [@Candicepan](https://github.com/Candicepan) 请问老师，要求ctr、cvr目标各位于一方，但是fit的时候要求label都位于同一个device，这样子的话算法逻辑应该是什么样子？如何进行设备间的传递？谢谢！

1. 需要改造 SLModel 支持 label 位于不同 device
2. data.to(device)
